util/unifycr/src/unifycr-cons.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2017, Lawrence Livermore National Security, LLC.
+ * Produced at the Lawrence Livermore National Laboratory.
+ *
+ * Copyright 2017, UT-Battelle, LLC.
+ *
+ * LLNL-CODE-741539
+ * All rights reserved.
+ *
+ * This is the license for UnifyCR.
+ * For details, see https://github.com/LLNL/UnifyCR.
+ * Please read https://github.com/LLNL/UnifyCR/LICENSE for full license text.
+ */
+
+/*
+ * Copyright (c) 2017, Lawrence Livermore National Security, LLC.
+ * Produced at the Lawrence Livermore National Laboratory.
+ * Copyright (c) 2017, Florida State University. Contributions from
+ * the Computer Architecture and Systems Research Laboratory (CASTL)
+ * at the Department of Computer Science.
+ *
+ * Written by: Hyogi Sim
+ * LLNL-CODE-728877. All rights reserved.
+ *
+ * This file is part of burstfs.
+ * For details, see https://github.com/llnl/burstfs
+ * Please read https://github.com/llnl/burstfs/LICENSE for full license text.
+ */
+
+/*
+ *
+ * Copyright (c) 2014, Los Alamos National Laboratory
+ *	All rights reserved.
+ *
+ */
+
+#ifndef _CONFIG_H
+#define _CONFIG_H
+#include <config.h>
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "unifycr.h"
+
+static char *consistency_strs[] = { "none", "laminated", "posix" };
+
+/**
+ * @brief return consistency enum entry matching with the given string.
+ *
+ * @param str input string of the consistency model
+ *
+ * @return matching enum entry on success, UNIFYCR_CM_INVALID (-1) otherwise
+ */
+unifycr_cm_t unifycr_read_consistency(const char *str)
+{
+    int i = 0;
+    unifycr_cm_t cons = UNIFYCR_CM_INVALID;
+
+    for (i = 0; i < sizeof(consistency_strs)/sizeof(char *); i++)
```

Comment:
`sizeof(consistency_strs)/sizeof(consistency_strs[0])` would be a more general construct. But you may need a shorter array name to fit it all on one line. Alternatively you could have a macro

#define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]))

since you use this construct below as well.

Reply:
I will update with the first method. Macro can potentially have a name conflict.